ksqldb-common/src/main/java/io/confluent/ksql/security/oauth/ClientSecretIdpConfig.java

```diff
@@ -0,0 +1,222 @@
+
+package io.confluent.ksql.security.oauth;
+
+import io.confluent.ksql.security.AuthType;
+import io.confluent.ksql.security.KsqlClientConfig;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.common.config.ConfigException;
+
+public final class ClientSecretIdpConfig implements IdpConfig {
+  private String idpTokenEndpointUrl = "";
+  private String idpClientId = "";
+  private String idpClientSecret = "";
+  private String idpScope = "";
+  private String idpScopeClaimName = "scope";
+  private String idpSubClaimName = "sub";
+  private Short idpCacheExpiryBufferSeconds = 300;
+
+  @Override
+  public AuthType getAuthType() {
+    return AuthType.OAUTHBEARER;
+  }
+
+  @Override
+  public String getAuthenticationMethod() {
+    return "CLIENT_SECRET";
+  }
+
+  @Override
+  public Map<String, Object> toIdpCredentialsConfig() {
+    final Map<String, Object> idpCredentialsConfig = new HashMap<>();
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_TOKEN_ENDPOINT_URL, idpTokenEndpointUrl);
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_CLIENT_ID, idpClientId);
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_CLIENT_SECRET, idpClientSecret);
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_SCOPE, idpScope);
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_SCOPE_CLAIM_NAME, idpScopeClaimName);
+    idpCredentialsConfig.put(KsqlClientConfig.BEARER_AUTH_SUB_CLAIM_NAME, idpSubClaimName);
+    idpCredentialsConfig.put(
+            KsqlClientConfig.BEARER_AUTH_CACHE_EXPIRY_BUFFER_SECONDS, idpCacheExpiryBufferSeconds);
+    return idpCredentialsConfig;
+  }
+
+  @Override
+  public void configure(final Map<String, ?> configs) {
+    this.idpTokenEndpointUrl =
+            (String) configs.get(KsqlClientConfig.BEARER_AUTH_TOKEN_ENDPOINT_URL);
+    this.idpClientId =
+            (String) configs.get(KsqlClientConfig.BEARER_AUTH_CLIENT_ID);
+    this.idpClientSecret =
+            (String) configs.get(KsqlClientConfig.BEARER_AUTH_CLIENT_SECRET);
+    this.idpScope = (String) configs.get(KsqlClientConfig.BEARER_AUTH_SCOPE);
+    if (configs.get(KsqlClientConfig.BEARER_AUTH_SCOPE_CLAIM_NAME) != null) {
+      this.idpScopeClaimName = (String) configs.get(KsqlClientConfig.BEARER_AUTH_SCOPE_CLAIM_NAME);
+    }
+    if (configs.get(KsqlClientConfig.BEARER_AUTH_SUB_CLAIM_NAME) != null) {
+      this.idpSubClaimName = (String) configs.get(KsqlClientConfig.BEARER_AUTH_SUB_CLAIM_NAME);
+    }
+    if (configs.get(KsqlClientConfig.BEARER_AUTH_CACHE_EXPIRY_BUFFER_SECONDS) != null) {
+      this.idpCacheExpiryBufferSeconds =
+              (Short) configs.get(KsqlClientConfig.BEARER_AUTH_CACHE_EXPIRY_BUFFER_SECONDS);
+    }
+    validate();
+  }
+
+  private void validate() throws ConfigException {
+    if (isNullOrEmpty(idpTokenEndpointUrl)) {
+      throw new ConfigException(
+      "Cannot configure OAuthbearer client credentials without token endpoint url");
```

Comment:
```suggestion
      "Cannot configure OAuthBearer client credentials without token endpoint url");
```
Let's make this change to the 2 exceptions which are below this one